skops/card/_model_card.py

```diff
@@ -1057,6 +1064,9 @@ def add_plot(
                 folded=folded,
             )
             self._add_single(section_name, section)
+
+            if path_in_repo is not None:
+                shutil.copy(plot_path, path_in_repo)
```

Comment:
this should be in the above section, and modify the `plot_path` variable accordingly.

Reply:
yes 👍🏼

Reply:
@adrinjalali Hmm, after thinking through this some I'm unsure if copying on model card save is the best. If we do it when a user calls `add_plot()` then we'd already have the path of the plot in the context. From what I understand of how `.save()` is working we'd have to loop through all the generated content, find the plots, and then save them.

That feels a little extra to me since we could just save the copy when a user adds the plot. What are your thoughts? Am I misunderstanding how `.save()` is working?

Reply:
You're right that `save` would do that in an iteration. But from the user's experience, to me it makes more sense. This way, users only provide a relative path to the destination of the model card, and at the end we save them all. It should also provide a more consistent experience since they can't mess up addresses and at the end not have them in the repo.

Reply:
I agree with Adrin that copying on `save` has some advantages that make it worth the extra effort. On top of what was said, it would help to avoid the problem of a user adding a plot, then overwriting the plot file, but that change not being reflected in the final model card.

Implementation-wise, I agree with Thomas that looping through all sections _again_ is a bit wasteful. However, I think this feature can be implemented without a second loop. For that, I would add an argument to `_generate_card` to optionally copy the figures. This argument is passed down to `_generate_content`, which can take care of the copying. This way, we don't iterate through all sections twice and avoid duplicating the logic (which can be problematic besides performance, e.g. the current implementation does not consider invisible sections).

Another implementation detail, instead of `if hasattr(section, "path")`, I would have probably done `if isinstance(section, PlotSection)`, but I'm open to be convinced otherwise :)